extensions/ql-vscode/src/contextual/astBuilder.ts

```diff
@@ -48,61 +48,75 @@ export default class AstBuilder {
 
     // Build up the parent-child relationships
     edgeTuples.tuples.forEach(tuple => {
-      const from = tuple[0] as EntityValue;
-      const to = tuple[1] as EntityValue;
-      const toId = to.id!;
-      const fromId = from.id!;
-
-      if (tuple[2] === 'semmle.order') {
-        astOrder.set(toId, Number(tuple[3]));
-      } else if (tuple[2] === 'semmle.label') {
-        childToParent.set(toId, fromId);
-        let children = parentToChildren.get(fromId);
-        if (!children) {
-          parentToChildren.set(fromId, children = []);
+      const [source, target, tupleType, orderValue] = tuple as [EntityValue, EntityValue, string, string];
+      const sourceId = source.id!;
+      const targetId = target.id!;
+
+      switch (tupleType) {
+        case 'semmle.order':
+          astOrder.set(targetId, Number(orderValue));
+          break;
+
+        case 'semmle.label': {
+          childToParent.set(targetId, sourceId);
+          let children = parentToChildren.get(sourceId);
+          if (!children) {
+            parentToChildren.set(sourceId, children = []);
+          }
+          children.push(targetId);
+          break;
         }
-        children.push(toId);
+
+        default:
+        // ignore other tupleTypes since they are not needed by the ast viewer
```

Comment:
Ah, ok, great --- I didn't know that other values of `tupleType` existed, and it's great to clarify here that we're explicitly ignoring them.

Reply:
Thanks for suggesting that change. I think it makes the code clearer.